pandas/io/sql.py

```diff
@@ -172,9 +172,15 @@ def _convert_arrays_to_dataframe(
     )
     if dtype_backend == "pyarrow":
         pa = import_optional_dependency("pyarrow")
-        arrays = [
-            ArrowExtensionArray(pa.array(arr, from_pandas=True)) for arr in arrays
-        ]
+        import pyarrow.compute as pc
+
+        result_arrays = []
+        for arr in arrays:
+            pa_array = pa.array(arr, from_pandas=True)
+            if arr.dtype == "string":
+                pa_array = pc.cast(pa_array, pa.string())
+            result_arrays.append(ArrowExtensionArray(pa_array))
```

Comment:
What's the reason for this cast? (and maybe add a comment about it)

Reply:
Yeah I'm still confused about this as well. `if arr.dtype == "string":` we are still casting to `pa.string()`? What would the result type of `pa.array(arr, from_pandas=True)`?

Reply:
Hm the comment above was incorrect, its like this:

We are now using large_string in our String Extension arrays, e.g. if you convert this to an ArrowExtensionArray it will also be large_string. This is inconsistent with the other I/O methods where ArrowExtensionArray is still pa.string, that's why I am casting it back here.

I am happy to change this as well, but rather in a follow up

Reply:
Ah okay that makes sense. I'm OK with this then but would be good to have a `# TODO` noting we may want to keep large_string here in the future